Library/Homebrew/extend/os/linux/dev-cmd/determine-test-runners.rb

```diff
@@ -0,0 +1,245 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "formula"
+
+class TestRunnerFormula
+  extend T::Sig
+
+  sig { returns(String) }
+  attr_reader :name
+
+  sig { returns(Formula) }
+  attr_reader :formula
+
+  sig { params(name: String).void }
+  def initialize(name)
+    @name = T.let(name, String)
+    @formula = T.let(Formula[name], Formula)
+    @dependent_hash = T.let({}, T::Hash[T::Boolean, T::Array[TestRunnerFormula]])
+    freeze
+  end
+
+  sig { returns(T::Boolean) }
+  def macos_only?
+    formula.requirements.any? { |r| r.is_a?(MacOSRequirement) && !r.version_specified? }
+  end
+
+  sig { returns(T::Boolean) }
+  def linux_only?
+    formula.requirements.any?(LinuxRequirement)
+  end
+
+  sig { returns(T::Boolean) }
+  def x86_64_only?
+    formula.requirements.any? { |r| r.is_a?(ArchRequirement) && (r.arch == :x86_64) }
+  end
+
+  sig { returns(T::Boolean) }
+  def arm64_only?
+    formula.requirements.any? { |r| r.is_a?(ArchRequirement) && (r.arch == :arm64) }
+  end
+
+  sig { returns(T.nilable(MacOSRequirement)) }
+  def versioned_macos_requirement
+    formula.requirements.find { |r| r.is_a?(MacOSRequirement) && r.version_specified? }
+  end
+
+  sig { params(macos_version: MacOS::Version).returns(T::Boolean) }
+  def compatible_with?(macos_version)
+    # Assign to a variable to assist type-checking.
```

Comment:
(FWIW this seems sensible regardless, since it involves fewer passes over `formula.requirements`, and the name is a bit more concise)